cirq-core/cirq/contrib/shuffle_circuits/shuffle_circuits_with_readout_benchmarking.py

```diff
@@ -17,23 +17,54 @@
 from __future__ import annotations
 
 import time
-from typing import TYPE_CHECKING
+from typing import Optional, Sequence, TYPE_CHECKING
 
+import attrs
 import numpy as np
+import sympy
 
-from cirq import circuits, ops, protocols, work
+from cirq import circuits, ops, protocols, study, work
+from cirq._compat import deprecated
 from cirq.experiments import SingleQubitReadoutCalibrationResult
 
 if TYPE_CHECKING:
     from cirq.study import ResultDict
 
 
-def _validate_input(
-    input_circuits: list[circuits.Circuit],
+@attrs.frozen
+class ReadoutBenchmarkingParams:
+    """Parameters for configuring readout benchmarking.
+
+    Attributes:
+        circuit_repetitions: The repetitions for `circuits`.
+        num_random_bitstrings: The number of random bitstrings for measuring readout.
+            If set to 0, no readout calibration circuits are generated.
+        readout_repetitions: The number of repetitions for each readout bitstring.
+    """
+
+    circuit_repetitions: int | list[int]
+    num_random_bitstrings: int = 100
```

Comment:
nit: in the future you can reduce the boilerplate code using attrs.validators, for example you can do
```suggestion
    num_random_bitstrings: int = attrs.field(default=100, validator=[attrs.validator.instance_of(int), attrs.validator.gt(0)])
```

Reply:
Sure! I will do this in my future prs!